### DISCUSSION TITLE
How do I manually show the "display field" from a foreign key in a table column

### DISCUSSION THREAD

**Discussion by kpturner** [Budibase/budibase#10226] [Help]
I have two postgreSQL tables:

**country**
id_country (autoincrement PK)
id_currency (foreign key to currency table)
label_country (selected as display field for table)

**currency**
id_currency (autoincrement PK)
label_currency (selected as display field for table)

I want to create a table/grid over the **country** table and I want one of the columns to be **currency.label_currency** field for the *country.id_currency* field.   

Now I know this is easily achieved by creating a relationship BUT I cannot do that because of this bug: https://github.com/Budibase/budibase/issues/10218

so my question is: how can I do this WITHOUT creating a relationship?  It is a sort of custom column where I need to override the display value with the result of a query or somesuch.

**Comment by andz-bb** [comment]
hey @kpturner 

sorry to hear you've been having issues with Budibase relationships. In this case could you create a custom query that joins the two tables manually? I tried it out with the following query:

```sql
SELECT country.id_country, country.label_country, currency.label_currency
FROM country
JOIN currency ON country.id_currency = currency.id_currency;
```

<img width="1262" alt="custom join query" src="https://user-images.githubusercontent.com/110921612/230343586-691a27dd-1572-4591-a90a-676dc23d5d75.png">

Hope this helps, if you have any further questions on this let me know ☺️

**Reply by kpturner** [reply]
Thanks - but where does one create this custom query?

**Reply by andz-bb** [reply]
Hi Kevin, to create the query go to your Data section, click on your Postgres datasource, scroll to the bottom and click the blue _Add query_ button.

<img width="1260" alt="image" src="https://user-images.githubusercontent.com/110921612/230367406-e79e343e-af07-40bb-a74b-aae71d54c667.png">


You can find more info on custom SQL queries in [our PostgreSQL documentation page here.](https://docs.budibase.com/docs/postgresql#better-approach-using-a-custom-query)

**Comment by kpturner** [comment]
Thankyou 😊On 6 Apr 2023, at 12:49, Andrew Thompson ***@***.***> wrote:﻿
Hi Kevin, to create the query go to your design section, click on your Postgres datasource, scroll to the bottom and click the blue Add query button.

You can find more info on custom SQL queries in our PostgreSQL documentation page here.

**Comment by kpturner** [comment]
Although the table works and shows the data as I expect, it doesn't behave in the same way that a table using a table based data source would.  By that I mean that all my other tables automatically refresh and show new records (or remove deleted records) if I navigate off to a modal to add/remove records then close the modal.  With this table, the data remains static until I force it to refresh.  Am I missing something?
